### ISSUE TITLE
[BUG] Loading  non-existent psd file causes assert

### ISSUE THREAD

**Issue by DougRogers** [OpenImageIO/oiio#2946]
This reproduces the exception.   

```

#include` <string>
#include <iostream>
#include <filesystem>
#include <OpenImageIO/imageio.h>
using namespace OIIO;


void main()
{

    std::string str("PRP_VillageDirtyRag_A.psd");

    auto in = ImageInput::open(str.c_str());
    if (in)
    {
        const ImageSpec &spec = in->spec();
        int xres              = spec.width;
        int yres              = spec.height;
        int channels          = spec.nchannels;
        // std::vector<unsigned char> pixels(xres * yres * channels);
        // in->read_image(TypeDesc::UINT8, &pixels[0]);
        in->close();
    }
}
```

![image](https://user-images.githubusercontent.com/7563851/114786248-a8273700-9d32-11eb-80f1-33cf53d895bb.png)

![image](https://user-images.githubusercontent.com/7563851/114786384-dc9af300-9d32-11eb-9f02-77febf9b7bf9.png)

**Comment by lgritz** [comment]
This looks suspiciously like #2886, which was fixed for release 2.2.13. Is it possible you are using a version of OIIO that does not incorporate this fix (2.2.12 or earlier, or built from master prior to approximately March 5?

**Comment by DougRogers** [comment]
Looks like a duplicate.  I am using  2.2.10, which is the lastest vcpkg version.
